tesseract/tesseract_environment/src/kdl/kdl_state_solver.cpp

```diff
@@ -218,21 +214,23 @@ void KDLStateSolver::calculateTransformsHelper(tesseract_common::TransformMap& t
     Eigen::Isometry3d local_frame, global_frame;
     KDLToEigen(current_frame, local_frame);
     global_frame = parent_frame * local_frame;
-    transforms[current_element.segment.getName()] = global_frame;
+    state.link_transforms[current_element.segment.getName()] = global_frame;
+    if (current_element.segment.getName() != scene_graph_->getRoot())
+      state.joint_transforms[current_element.segment.getJoint().getName()] = global_frame;
```

Comment:
What does this do?

Reply:
perhaps that should go in the detailed description of the docs?

Reply:
Does this set both the link transform and joint transform to the same transformation (`global_frame`)?

Reply:
Yea, so a child link of a joint has the same transform as the joint.